<?xml version="1.0" ?>
<launch>
  <arg name="gui" default="true" />
  <arg name="world_name" default="worlds/empty.world"/> <!-- Note: the world_name is with respect to GAZEBO_RESOURCE_PATH environmental variable, but can also be an absolute path -->

  <arg name="robot_x"   default="0.0" />
  <arg name="robot_y"   default="0.0" />
  <arg name="robot_yaw" default="0.0" />

  


  <group>
   
    <remap from="/joint_states"                   to="/mir/joint_states" />
    <remap from="/mobile_base_controller/cmd_vel" to="/cmd_vel" />
    <remap from="/mobile_base_controller/odom"    to="/odom_comb" />
    
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="world_name" value="$(arg world_name)"/>
      <arg name="paused" value="false" />
      <arg name="gui" value="$(arg gui)" />
    </include>
  </group>

  <!-- spawn robot and bring up controllers etc. -->
  
  <include file="$(find robot_controler)/launch/mir_gazebo_common.launch">
    <arg name="robot_x"   value="$(arg robot_x)" />
    <arg name="robot_y"   value="$(arg robot_y)" />
    <arg name="robot_yaw" value="$(arg robot_yaw)" />
  </include>
  <node name="TestController" pkg="robot_controler" type="class_controler.py"> </node>
</launch>
